Build failure with C++20 mode on Visual Studio 2019 16.3 Preview 4.0

FAILED: src/CMakeFiles/pretty_grades.dir/pretty_grades.cpp.obj
C:\PROGRA~2\MIB055~1\2019\Preview\VC\Tools\MSVC\1423~1.281\bin\Hostx86\x86\cl.exe  /nologo /TP  -IC:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include /DWIN32 /D_WINDOWS /GR /EHsc /Zi /Ob0 /Od /RTC1 -MDd   /W4 /WX /D_USE_MATH_DEFINES /permissive- /experimental:preprocessor /wd5105 -std:c++latest /showIncludes /Fosrc\CMakeFiles\pretty_grades.dir\pretty_grades.cpp.obj /Fdsrc\CMakeFiles\pretty_grades.dir\ /FS -c C:\Users\fiucha\source\repos\cppcon-ranges\src\pretty_grades.cpp
C:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include\range/v3/view/tail.hpp(84): error C2760: syntax error: unexpected token '(', expected 'expression'
C:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include\range/v3/view/tail.hpp(103): note: see reference to class template instantiation 'ranges::tail_view<Rng>' being compiled
C:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include\range/v3/view/tail.hpp(84): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include\range/v3/view/tail.hpp(84): error C2059: syntax error: '('
C:\Users\fiucha\source\repos\cppcon-ranges\external\lib\ranges\include\range/v3/view/tail.hpp(84): error C2059: syntax error: '('
ninja: build stopped: subcommand failed.



The same code builds fine with "-std:c++17" instead of "-std:c++latest" and runs.

--- a/include/range/v3/iterator/concepts.hpp
+++ b/include/range/v3/iterator/concepts.hpp
@@ -122,7 +122,7 @@ namespace ranges
     (
         template(typename I)
         concept readable,
-            requires (uncvref_t<I> const i) (
+            requires (uncvref_t<I> const& i) (
                 // { *i } -> same_as<iter_reference_t<I>>;
                 // { iter_move(i) } -> same_as<iter_rvalue_reference_t<I>>;
                 concepts::requires_<same_as<decltype(*i), iter_reference_t<I>>>,

--- a/include/range/v3/iterator/concepts.hpp
+++ b/include/range/v3/iterator/concepts.hpp
@@ -122,7 +122,7 @@ namespace ranges
     (
         template(typename I)
         concept readable,
-            requires (uncvref_t<I> const& i) (
+            requires (uncvref_t<I> const &i) (
                 // { *i } -> same_as<iter_reference_t<I>>;
                 // { iter_move(i) } -> same_as<iter_rvalue_reference_t<I>>;
                 concepts::requires_<same_as<decltype(*i), iter_reference_t<I>>>,

--- a/include/range/v3/view/tail.hpp
+++ b/include/range/v3/view/tail.hpp
@@ -81,7 +81,7 @@ namespace ranges
         }
         // Strange cast to bool in the requires clause is to work around gcc bug.
         CPP_member
-        constexpr auto CPP_fun(size)()(requires bool(sized_range<Rng>))
+        constexpr auto CPP_fun(size)()(requires (bool(sized_range<Rng>)))
         {
             using size_type = range_size_t<Rng>;
             return range_cardinality<Rng>::value >= 0
@@ -89,7 +89,7 @@ namespace ranges
                        : detail::prev_or_zero_(ranges::size(rng_));
         }
         CPP_member
-        constexpr auto CPP_fun(size)()(const requires bool(sized_range<Rng const>))
+        constexpr auto CPP_fun(size)()(const requires (bool(sized_range<Rng const>)))
         {
             using size_type = range_size_t<Rng>;
             return range_cardinality<Rng>::value >= 0

